### PR TITLE
msg: move DispatchStrategy into test/

### PR DESCRIPTION
**Pull request by Rethan**
Since commit cc9a914, xio related code were all removed. Therefore only direct messenger in test will use "dispatch strategy" things,  so remove to there may simplify src/msg.

Signed-off-by: haoyixing <haoyixing@kuaishou.com>

"""
-->
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/test/direct_messenger/test_direct_messenger.cc [hunk 1]
```diff
@@ -11,8 +11,8 @@
 #include "common/ceph_argparse.h"
 
 #include "DirectMessenger.h"
-#include "msg/FastStrategy.h"
-#include "msg/QueueStrategy.h"
+#include "FastStrategy.h"
+#include "QueueStrategy.h"
```
Comment (tchaikov) [inline src/test/direct_messenger/test_direct_messenger.cc:15]: might want to update `src/msg/CMakeLists.txt` and `src/test/direct_messenger/CMakeLists.txt` accordingly.

Comment (Rethan) [reply src/test/direct_messenger/test_direct_messenger.cc:15]: I have made some changes, but direct messenger don’t seem to work properly even if uncomment cmake statements.(It was commented out long ago by sage at d290ffc098 due to no users and failing). Maybe we can cleanup direct_messenger in the future.

Comment (tchaikov) [reply src/test/direct_messenger/test_direct_messenger.cc:15]: @Rethan sure. what i really wanted was to remove `QueueStrategy.cc` from `msg/CMakeLists.txt`. and it's addressed in your latest revision. so all good!

**Review by tchaikov — APPROVED** [review]

**Review by tchaikov — CHANGES_REQUESTED** [review]
@Rethan could you squash these two commits into a single one and change the title to something like

> msg: move DispatchStrategy into test/
>
> <and expand on the reason a little bit>

**Comment by Rethan** [comment]
> @Rethan could you squash these two commits into a single one and change the title to something like
> 
> > msg: move DispatchStrategy into test/
> > # and please expand on the reason a little bit

Thanks, working on it.

**Review by tchaikov — APPROVED** [review]

**Comment by tchaikov** [comment]
https://shaman.ceph.com/builds/ceph/wip-kefu-testing-2020-10-23-1631/859eb5d2de58a5fffc9e4e725f09895d3b5a79c0/

**Pull request merged**